src/PolygonWorldID.sol

```diff
@@ -26,7 +42,88 @@ contract PolygonWorldID is WorldIDBridge, FxBaseChildTunnel, Ownable {
     constructor(uint8 _treeDepth, address _fxChild)
         WorldIDBridge(_treeDepth)
         FxBaseChildTunnel(_fxChild)
-    {}
+    {
+        receiveRootSelector = bytes4(keccak256("receiveRoot(uint256,uint128)"));
+        receiveRootHistoryExpirySelector = bytes4(keccak256("setRootHistoryExpiry(uint256)"));
+    }
+
+    /// @notice grabSelector, takes a byte array _payload as input and returns the first 4 bytes
+    /// of the array as a bytes4 value _selector. The function uses EVM assembly language
+    /// to load the 4-byte selector from the _payload array and then shift it left by 224 bits
+    /// (0xE0 in hexadecimal) to get the correct value.
+    /// @param _payload The byte array from which to extract the selector
+    /// @return _selector The first 4 bytes of the _payload array (the function selector from encodeWithSignature)
+    function grabSelector(bytes memory _payload) internal pure returns (bytes4 _selector) {
```

Comment:
This is missing a check. You need to ensure that the `_payload` is >= 4 bytes long, otherwise we _can't_ properly grab the selector.

Reply:
Also, this function should be pulled out into a library (still internal) so that it can be tested independently.